src/generated/xinput.rs

```diff
@@ -103,7 +97,7 @@ where
     let extension_information = conn.extension_information(X11_EXTENSION_NAME)?
         .ok_or(ConnectionError::UnsupportedExtension)?;
     let length_so_far = 0;
-    let name_len: u16 = name.len().try_into()?;
+    let name_len = name.len() as u16;
```

Comment:
I think this needs to be changed to the old version

Reply:
Should it used `?` or `.unwrap()` (or `.expect()`)?

Note that there are other places where `as` is used in the current implementation, such as:

https://github.com/psychon/x11rb/blob/9107c132d703d8c6dc69c9ea3956c05af6373f7c/src/generated/xkb.rs#L3559-L3579

In this case, `?` could not be used because `Serialize` cannot return errors.

Reply:
I'd say it should be `?` here, since it is not an error in the generated code itself, but with the caller passing in something that is too big. This case ends up as a `ParseError`, which is not that useful, but better than nothing:
https://github.com/psychon/x11rb/blob/787b0955bbf73ecd542f0929a3d48102d63eb587/src/errors.rs#L21-L25
https://github.com/psychon/x11rb/blob/787b0955bbf73ecd542f0929a3d48102d63eb587/src/errors.rs#L167-L171

> Note that there are other places where as is used in the current implementation, such as:

Hm... yeah... I guess there is not much that can be done about this here. Dunno if it would be better to panic instead of generating a broken serialisation. Let's hope we never have to debug such a problem. (So I guess a panic really would be better :-) )

Reply:
Fixed, but it now uses `.expect()` instrad of `?`.

Reply:
Fine with me